dspace-server-webapp/src/test/java/org/dspace/app/rest/ResourcePolicyRestRepositoryIT.java

```diff
@@ -0,0 +1,578 @@
+                .withUser(eperson1).build();
+
+        ResourcePolicy resourcePolicyOfCollection = ResourcePolicyBuilder.createResourcePolicy(context)
+                .withDspaceObject(collection).withAction(Constants.REMOVE)
+                .withPolicyType(ResourcePolicy.TYPE_SUBMISSION).withUser(eperson1).build();
+
+        context.restoreAuthSystemState();
+
+        String authToken = getAuthToken(eperson1.getEmail(), "qwerty01");
+
+        getClient(authToken)
+                .perform(get("/api/authz/resourcepolicies/search/eperson")
+                        .param("uuid", eperson1.getID().toString())
+                        .param("resource", community.getID().toString()))
+                .andExpect(status().isOk()).andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$._embedded.resourcepolicies",Matchers.containsInAnyOrder(
+                        ResoucePolicyMatcher.matchResourcePolicy(resourcePolicyOfCommunity),
+                        ResoucePolicyMatcher.matchResourcePolicy(secondResourcePolicyOfCommunity)
+                        )))
+                .andExpect(jsonPath("$._embedded.resourcepolicies",
+                        Matchers.not(is(ResoucePolicyMatcher.matchResourcePolicy(resourcePolicyOfCollection)))))
+                .andExpect(jsonPath("$._links.self.href",
+                        Matchers.containsString("api/authz/resourcepolicies/search/eperson")))
+                .andExpect(jsonPath("$.page.totalElements", is(2)));
+    }
+
+    /**
+     * this method return 422, but must return 400
+     * FIX BUG
+     * @throws Exception
+     */
+    @Test
+    public void searchResoucesPolicyEPersonWithoutParametersBadRequestTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson eperson1 = EPersonBuilder.createEPerson(context).withEmail("myemail@mail.com").withPassword("qwerty01")
+                .build();
+        context.restoreAuthSystemState();
+
+        String authToken = getAuthToken(eperson1.getEmail(), "qwerty01");
+        getClient(authToken).perform(get("/api/authz/resourcepolicies/search/eperson"))
+                .andExpect(status().isUnprocessableEntity());
+    }
+
+    @Test
+    public void searchResoucesPolicyByEPersonUuidUnAuthenticatedTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson eperson1 = EPersonBuilder.createEPerson(context)
+                .withEmail("myemail@mail.com")
+                .withPassword("qwerty01")
+                .build();
+
+        Community community = CommunityBuilder.createCommunity(context).withName("My community").build();
+
+        ResourcePolicy resourcePolicy = ResourcePolicyBuilder.createResourcePolicy(context)
+                .withDspaceObject(community)
+                .withAction(Constants.READ)
+                .withPolicyType(ResourcePolicy.TYPE_CUSTOM)
```

Comment:
also resource policies of communities usually haven't a policy type, please remove